pkgs/by-name/ne/nekoray/package.nix

```diff
@@ -110,22 +111,30 @@ stdenv.mkDerivation (finalAttrs: {
   installPhase = ''
     runHook preInstall
 
-    install -Dm755 nekoray "$out/share/nekoray/nekoray"
-    mkdir -p "$out/bin"
-    ln -s "$out/share/nekoray/nekoray" "$out/bin"
+    ${if stdenv.isLinux then ''
+      install_dir="$out/share/nekoray"
+
+      install -Dm755 nekoray "$out/share/nekoray/nekoray"
+      mkdir -p "$out/bin"
+      ln -s "$out/share/nekoray/nekoray" "$out/bin"
+      install -Dm644 "$src/res/public/nekoray.png" "$out/share/icons/hicolor/256x256/apps/nekoray.png"
+    '' else ''
+      install_dir="$out/Applications/nekoray.app/Contents/MacOS"
+
+      mkdir -p $out/Applications
+      cp -r nekoray.app $out/Applications
```

Comment:
Maybe we could also symlink it back into $out/bin, since `nix run` will try to run the executable from there.

*(If the symlinking breaks it, try to use `makeWrapper` instead)*

Reply:
regarding this - i'm not entirely sure this is a good idea
most gui applications don't need to be in path, and if i symlink it to `bin/` it will inevitably end up there :c
it's somewhat of a necessary evil on linux, but on macos we have `/Applications`

to support `nix run`-ning maybe it's a better idea to do something like this?
```diff
@@ -121,7 +124,7 @@ stdenv.mkDerivation (finalAttrs: {
     '' else ''
       install_dir="$out/Applications/nekoray.app/Contents/MacOS"
 
-      mkdir -p $out/Applications
+      mkdir -p $out/Applications $out/bin
       cp -r nekoray.app $out/Applications
     ''}
 
@@ -164,7 +167,7 @@ stdenv.mkDerivation (finalAttrs: {
     description = "Qt based cross-platform GUI proxy configuration manager";
     homepage = "https://github.com/MatsuriDayo/nekoray";
     license = lib.licenses.gpl3Plus;
-    mainProgram = "nekoray";
+    mainProgram = if stdenv.isDarwin then "../Applications/nekoray.app/Contents/MacOS/nekoray" else "nekoray";
     maintainers = with lib.maintainers; [ tomasajt ];
   };
 })
```

though it definitely looks a bit hacky 🥴